test/Dialect/Linalg/transform-patterns.mlir

```diff
@@ -315,3 +315,53 @@ func @matmul_perm(%A: memref<?x?xf32, offset: ?, strides: [?, 1]>,
 // CHECK      :                 loop.for {{.*}} = %[[c0]] to {{.*}} step %[[c30]] {
 // CHECK      :                   loop.for {{.*}} = %[[c0]] to {{.*}} step %[[c40]] {
 // CHECK      :                           linalg.matmul({{.*}}, {{.*}}, {{.*}}) : memref<?x?xf32, #[[STRIDED_2D]]>, memref<?x?xf32, #[[STRIDED_2D]]>, memref<?x?xf32, #[[STRIDED_2D]]>
+
+func @promote_subview_matmul(%arg0: memref<?x?xf32, offset: ?, strides: [?, 1]>,
+                             %arg1: memref<?x?xf32, offset: ?, strides: [?, 1]>,
+                             %arg2: memref<?x?xf32, offset: ?, strides: [?, 1]>) {
+  %c2000 = constant 2000 : index
+  %c3000 = constant 3000 : index
+  %c4000 = constant 4000 : index
+  %c0 = constant 0 : index
+  %c1 = constant 1 : index
+  %0 = dim %arg0, 0 : memref<?x?xf32, offset: ?, strides: [?, 1]>
+  %1 = dim %arg0, 1 : memref<?x?xf32, offset: ?, strides: [?, 1]>
+  %2 = dim %arg1, 1 : memref<?x?xf32, offset: ?, strides: [?, 1]>
+  loop.for %arg3 = %c0 to %0 step %c2000 {
+    loop.for %arg4 = %c0 to %2 step %c3000 {
+      loop.for %arg5 = %c0 to %1 step %c4000 {
+        %3 = std.subview %arg0[%arg3, %arg5][%c2000, %c4000][%c1, %c1] :
+             memref<?x?xf32, offset: ?, strides: [?, 1]> to memref<?x?xf32, offset: ?, strides: [?, ?]>
```

Comment:
The subview result type can keep the same strides as the original type.

Reply:
When we tried that (`strides : [?,1]` for the output of the  `subview`), I got the error:
`transform-patterns.mlir:333:14: error: expected result type to have dynamic strides` 
Am I missing something ?

Reply:
THe current specification of subview op doesnt allow mixing and matching static and dynamic strides. Either they should be all static or all dynamic.  So you cannot have [?, ?].
In this case even though the stride operands are constants, the base memref has dynamic stride. So the subview op result must have all strides dynamic.
To fix this, the specification of subview op has to change to allow for this more general representation.

Reply:
ah yes, forgot about that :) thanks!